src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int capacity;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        this.capacity = table.length;
+    }
 
     @Override
     public void put(K key, V value) {
+        int threshold = (int) (capacity * LOAD_FACTOR);
+        if (size > threshold) {
+            resize();
+        }
+
+        int bucketIndex = getIndex(key);
+        if (table[bucketIndex] != null) {
+            Node<K, V> currentNode = table[bucketIndex];
 
+            while (currentNode != null) {
+                if (areKeysEqual(key, currentNode.key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+
+        Node<K, V> newNode = new Node<>(key, value);
+        addNode(table, newNode);
```

Comment:
what we need to do here is to assign a new node after we have iterated through the previous ones and haven't found the one we need, but without another `while` loop that we now have in `addNode()`